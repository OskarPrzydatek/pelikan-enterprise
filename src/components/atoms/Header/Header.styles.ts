import styled from 'styled-components';

export const Header = styled.h1`
  margin: 0;
  font-family: ${({ theme }) => theme.fonts.main};
  color: ${({ theme }) => theme.colors.darkBlue};

  font-size: ${({ theme }) => theme.fontSizes.s};
  margin: ${({ theme }) => theme.spaces.xs} 0;

  @media ${({ theme }) => theme.breakpoints.phone} {
    font-size: ${({ theme }) => theme.fontSizes.m};
    margin: ${({ theme }) => theme.spaces.xs} 0;
  }

  @media ${({ theme }) => theme.breakpoints.tablet} {
    font-size: ${({ theme }) => theme.fontSizes.l};
    margin: ${({ theme }) => theme.spaces.s} 0;
  }

  @media ${({ theme }) => theme.breakpoints.desktop} {
    font-size: ${({ theme }) => theme.fontSizes.l};
    margin: ${({ theme }) => theme.spaces.s} 0;
  }
`;
