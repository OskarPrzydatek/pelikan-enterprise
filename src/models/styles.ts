import {
  DefaultTheme,
  FlattenInterpolation,
  ThemeProps,
} from 'styled-components';

export interface ICSS {
  css?: FlattenInterpolation<ThemeProps<DefaultTheme>>;
}
