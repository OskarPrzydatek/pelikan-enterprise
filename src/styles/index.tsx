export * from './layouts';
export { theme } from './theme';
