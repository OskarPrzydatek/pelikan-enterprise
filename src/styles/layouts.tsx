import styled, { createGlobalStyle } from 'styled-components';

export const GlobalStyleLayout = createGlobalStyle`
  * {
		box-sizing: border-box;
    outline: none;
	} 

  body {
    margin: 0;
    padding: 0;
  }

  #root {
    height: 100vh;
    display: grid;
    grid-template-rows: .5fr 8fr .5fr;
  }

	a {
		color: inherit;
		text-decoration: none;
	}

	ul {
		padding: 0;
    margin: 0;
		list-style-type: none;
	}
`;

export const MainLayout = styled.main`
  padding: ${({ theme }) => theme.spaces.l};
`;

export const PageLayout = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;
