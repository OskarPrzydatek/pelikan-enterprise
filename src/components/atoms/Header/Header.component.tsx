import { IComponent } from '~/models';

import * as S from './Header.styles';

interface IHeder extends IComponent {
  children: React.ReactNode;
}

export const Header: React.FC<IHeder> = ({ children, dataTestID }: IHeder) => {
  return <S.Header data-testid={dataTestID}>{children}</S.Header>;
};
