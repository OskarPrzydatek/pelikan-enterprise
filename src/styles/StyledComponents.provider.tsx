import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from './layouts';
import { theme } from './theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};
