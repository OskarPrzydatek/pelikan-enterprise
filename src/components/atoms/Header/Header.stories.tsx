import { Meta, StoryObj } from '@storybook/react';

import { StyledComponentsProvider } from '~/styles';

import { Header } from './Header.component';

const meta: Meta<typeof Header> = {
  title: 'Atoms/Header',
  component: Header,
};

export default meta;
type Story = StoryObj<typeof Header>;

export const HeaderStory: Story = {
  render: () => (
    <StyledComponentsProvider>
      <Header>String Header</Header>
    </StyledComponentsProvider>
  ),
};
