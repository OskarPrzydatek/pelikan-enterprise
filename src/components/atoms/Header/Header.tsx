import { IComponent } from '~/models';

import * as S from './Header.styles';

interface IHeder extends IComponent {
  children: React.ReactNode;
}

export const Header: React.FC<IHeder> = ({
  css,
  children,
  dataTestID,
}: IHeder) => {
  return (
    <S.Header css={css} data-testid={dataTestID}>
      {children}
    </S.Header>
  );
};
