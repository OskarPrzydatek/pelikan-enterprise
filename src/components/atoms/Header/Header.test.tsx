import { render, screen } from '@testing-library/react';
import { describe, test } from 'vitest';

import { StyledComponentsProvider } from '~/styles';

import { Header } from './Header.component';

describe('Header', () => {
  test('component snapshot', () => {
    const view = render(
      <StyledComponentsProvider>
        <Header dataTestID="header-atom">String Header</Header>
      </StyledComponentsProvider>
    );
    expect(view).toMatchSnapshot();
  });

  test('ensure header children renders correctly', () => {
    render(
      <StyledComponentsProvider>
        <Header dataTestID="header-atom">String Header</Header>
      </StyledComponentsProvider>
    );
    const header = screen.getByTestId('header-atom');
    expect(header).toHaveTextContent('String Header');
  });
});
