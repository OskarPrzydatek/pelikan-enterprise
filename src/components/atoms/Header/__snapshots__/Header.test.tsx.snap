// Vitest Snapshot v1

exports[`Header > component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  margin: 0;
  font-family: 'Nanum Gothic',sans-serif;
  color: #21313F;
  font-size: 1.4rem;
  margin: .5% 0;
}

@media (min-width:390px) {
  .c0 {
    font-size: 1.8rem;
    margin: .5% 0;
  }
}

@media (min-width:834px) {
  .c0 {
    font-size: 2rem;
    margin: .7% 0;
  }
}

@media (min-width:1728px) {
  .c0 {
    font-size: 1rem;
    margin: .7% 0;
  }
}

<body>
    <div>
      <h1
        class="c0"
        data-testid="header-atom"
      >
        String Header
      </h1>
    </div>
  </body>,
  "container": .c0 {
  margin: 0;
  font-family: 'Nanum Gothic',sans-serif;
  color: #21313F;
  font-size: 1.4rem;
  margin: .5% 0;
}

@media (min-width:390px) {
  .c0 {
    font-size: 1.8rem;
    margin: .5% 0;
  }
}

@media (min-width:834px) {
  .c0 {
    font-size: 2rem;
    margin: .7% 0;
  }
}

@media (min-width:1728px) {
  .c0 {
    font-size: 1rem;
    margin: .7% 0;
  }
}

<div>
    <h1
      class="c0"
      data-testid="header-atom"
    >
      String Header
    </h1>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
